### PR TITLE
using dtl logger for dotnetcore build

### PR DESCRIPTION
**Pull request by tejasd1990**

### CODE REVIEW

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 1]
```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
...
+            if(this.isBuildCommand()) {
+                var detailId = "";
+                var projectFilePath = path.dirname(projectFile);
+                if(this.logProjectEvents) {
+                    detailId = uuidV4();
+                    var projectFileName = path.basename(projectFile);
+                    var detailName = tl.loc("Build", projectFileName);
+                    var startTime = new Date(Date.now()).toISOString();
+                    //Todo: can we use named parameters in typescipt?
+                    tl.logDetail(detailId, "", "", "Process", detailName, 0, startTime, "", 0, tl.TaskState.Initialized);
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: what does this line do?

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 2]
```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
...
+                }
+                var loggerAssembly = path.join(__dirname, 'dotnet-build-helpers/Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll');
+                dotnet.arg(`-dl:CentralLogger,\"${loggerAssembly}\";\"RootDetailId=${detailId}|SolutionDir=${projectFilePath}\"*ForwardingLogger,\"${loggerAssembly}\"`);
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Do we need anything other than CentralLogger?

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts (hunk 1 above)
Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: This initializes the [timeline](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#logdetail-create-and-update-a-timeline-record-for-a-task). Timeline records work as parent-children. We need to initialize the first timeline record. Subsequent timelines are children of the existing timeline records, which already exist.

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts (hunk 2 above)
Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: The same logger is used as ForwardingLogger as well

File: Tasks/DotNetCoreCLIV2/Tests/L0.ts [hunk 3]
```diff
@@ -205,18 +205,6 @@ describe('DotNetCoreExe Suite', function () {
         done();
     });
 
-    it('build passes when zero match found with empty string', (done: MochaDone) => {
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/Tests/L0.ts]: Why is the test removed?

**Comment by bishal-pdMSFT** [comment]
Increment task version

File: Tasks/DotNetCoreCLIV2/Tests/L0.ts (hunk 3 above)
Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/Tests/L0.ts]: It is hardcoded for responses to string constants for the various dotnet commands like "dotnet build", "dotnet run" etc. In this case, the command string will contain additional parameter "_`-dl:CentralLogger,\"${loggerAssembly}\"*ForwardingLogger,\"${loggerAssembly}\"`_", which is not a constant and depends on the path of executoin. So,I have removed it for now. Will see how to configure it and add it back.

**Comment by bishal-pdMSFT** [comment]
@tejasd1990 please port this change to releases/M165 and also update external dependencies

**Pull request merged**